Add detailed example for rethrow to explainer (#263)
WebAssembly/exception-handling: proposals/exception-handling/Exceptions.md

The `rethrow` here references `try $l2`, but the `rethrow` is not within its
`catch` block.

> The below is an example that includes all the cases explained. The numbers

Reviewer: ```suggestion
The example below includes all of the cases explained above. The numbers
```
